### PR TITLE
feat(cce): support pod_security_groups in node pool

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

support pod_security_groups in node pool

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
support pod_security_groups in node pool
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccCCENodePool_podSecurityGroups'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccCCENodePool_podSecurityGroups -timeout 360m -parallel 4
=== RUN   TestAccCCENodePool_podSecurityGroups
=== PAUSE TestAccCCENodePool_podSecurityGroups
=== CONT  TestAccCCENodePool_podSecurityGroups
--- PASS: TestAccCCENodePool_podSecurityGroups (941.41s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       941.492s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_node_pool_test.go [hunk 1]
```diff
@@ -403,6 +429,56 @@ resource "huaweicloud_cce_node_pool" "test" {
 `, testAccCCENodePool_Base(rName), rName, rName)
 }
 
+func testAccCCENodePool_podSecurityGroups(rName string) string {
+	return fmt.Sprintf(`
+%s
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_node_pool_test.go]:
```suggestion
%[1]s
```

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_node_pool_test.go [hunk 2]
```diff
@@ -403,6 +429,56 @@ resource "huaweicloud_cce_node_pool" "test" {
...
+
+data "huaweicloud_availability_zones" "test" {}
+
+resource "huaweicloud_compute_keypair" "test" {
+  name = "%[2]s"
+  public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDAjpC1hwiOCCmKEWxJ4qzTTsJbKzndLo1BCz5PcwtUnflmU+gHJtWMZKpuEGVi29h0A/+ydKek1O18k10Ff+4tyFjiHDQAT9+OfgWf7+b1yK+qDip3X1C0UPMbwHlTfSGWLGZquwhvEFx9k3h/M+VtMvwR1lJ9LUyTAImnNjWG7TAIPmui30HvM2UiFEmqkr4ijq45MyX2+fLIePLRIFuu1p4whjHAQYufqyno3BS48icQb4p6iVEZPo4AE2o9oIyQvj2mx4dk5Y8CgSETOZTYDOR3rU2fZTRDRgPJDH9FWvQjF5tA0p3d9CoWWd2s6GKKbfoUIi8R/Db1BSPJwkqB jrp-hp-pc"
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_node_pool_test.go]:
```suggestion
resource "huaweicloud_compute_keypair" "test" {
  name       = "%[2]s"
  public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDAjpC1hwiOCCmKEWxJ4qzTTsJbKzndLo1BCz5PcwtUnflmU+gHJtWMZKpuEGVi29h0A/+ydKek1O18k10Ff+4tyFjiHDQAT9+OfgWf7+b1yK+qDip3X1C0UPMbwHlTfSGWLGZquwhvEFx9k3h/M+VtMvwR1lJ9LUyTAImnNjWG7TAIPmui30HvM2UiFEmqkr4ijq45MyX2+fLIePLRIFuu1p4whjHAQYufqyno3BS48icQb4p6iVEZPo4AE2o9oIyQvj2mx4dk5Y8CgSETOZTYDOR3rU2fZTRDRgPJDH9FWvQjF5tA0p3d9CoWWd2s6GKKbfoUIi8R/Db1BSPJwkqB jrp-hp-pc"
}
```

**Comment by Lance52259** [comment]
We can obtain the security groups via node pool API:
GET /api/v3/projects/{project_id}/cluster/{cluster_id}/nodepools/{nodepool_id}
And we can save this configuration to the state file.

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_node_pool_test.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_node_pool_test.go]: fixed

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_node_pool_test.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_node_pool_test.go]: fixed

**Comment by Jason-Zhang9309** [comment]
> We can obtain the security groups via node pool API: GET /api/v3/projects/{project_id}/cluster/{cluster_id}/nodepools/{nodepool_id} And we can save this configuration to the state file.

this API is already used in Read func, some parameters not returned, I've add a comment about this

**Pull request merged**